spyder/plugins/plots/widgets/tests/test_plots_widgets.py

```diff
@@ -210,6 +211,21 @@ def test_close_all_figures(figbrowser, tmpdir, fmt):
     assert figbrowser.thumbnails_sb.get_current_index() == -1
     assert figbrowser.thumbnails_sb.current_thumbnail is None
     assert figbrowser.figviewer.figcanvas.fig is None
+    assert len(figbrowser.thumbnails_sb.findChildren(FigureThumbnail)) == 0
+
+
+@pytest.mark.parametrize("fmt", ['image/png', 'image/svg+xml'])
+def test_close_one_thumbnail(figbrowser, tmpdir, fmt):
+    """
+    Test the thumbnail is removed from the GUI.
+    """
+    # Add two figures to the browser
+    add_figures_to_browser(figbrowser, 2, tmpdir, fmt)
+    assert len(figbrowser.thumbnails_sb.findChildren(FigureThumbnail)) == 2
+
+    # Remove the first figure
+    figbrowser.close_figure()
+    assert len(figbrowser.thumbnails_sb.findChildren(FigureThumbnail)) == 1
```

Comment:
Hi @steff456 great work :-).

Just to check, did you tests this test without your fix and checked it actually failed?

Otherwise we should call the actual method that the `X` button calls. That method is `remove_thumbnail`.

Reply:
Let me check if the test doesn't work without the fix and I will tell you. But I think maybe it is better to call the `remove_thumbnail` method.